examples/userapi/05_conditional_dimension.ipynb

```diff
@@ -767,6 +783,280 @@
+      "\n",
+      "struct dataobj\n",
+      "{\n",
+      "  void *restrict data;\n",
+      "  int * size;\n",
+      "  int * npsize;\n",
+      "  int * dsize;\n",
+      "  int * hsize;\n",
+      "  int * hofs;\n",
+      "  int * oofs;\n",
+      "} ;\n",
+      "\n",
+      "struct profiler\n",
+      "{\n",
+      "  double section0;\n",
+      "} ;\n",
+      "\n",
+      "\n",
+      "int Kernel(struct dataobj *restrict f_vec, struct dataobj *restrict g_vec, const int x_M, const int x_m, const int y_M, const int y_m, struct profiler * timers)\n",
+      "{\n",
+      "  float (*restrict f)[f_vec->size[1]] __attribute__ ((aligned (64))) = (float (*)[f_vec->size[1]]) f_vec->data;\n",
+      "  int (*restrict g) __attribute__ ((aligned (64))) = (int (*)) g_vec->data;\n",
+      "\n",
+      "  /* Flush denormal numbers to zero in hardware */\n",
+      "  _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);\n",
+      "  _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);\n",
+      "\n",
+      "  /* Begin section0 */\n",
+      "  START_TIMER(section0)\n",
+      "  for (int x = x_m; x <= x_M; x += 1)\n",
+      "  {\n",
+      "    for (int y = y_m; y <= y_M; y += 1)\n",
+      "    {\n",
+      "      if (f[x][y] != 0)\n",
+      "      {\n",
+      "        int r0 = 1;\n",
+      "        g[1] += r0;\n",
+      "      }\n",
+      "    }\n",
+      "  }\n",
+      "  STOP_TIMER(section0,timers)\n",
+      "  /* End section0 */\n",
+      "\n",
+      "  return 0;\n",
+      "}\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "10"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "#NBVAL_IGNORE_OUTPUT\n",
+    "# ConditionalDimension with Ne(f, 0) condition\n",
+    "ci = ConditionalDimension(name='ci', parent=f.dimensions[-1],\n",
+    "                          condition=Ne(f, 0))\n",
+    "\n",
+    "# g is our counter. g needs to be a Function to write in\n",
+    "g = Function(name='g', shape=(1,), dimensions=(ci,), dtype=np.int32)\n",
+    "\n",
+    "# Extra Scalar used to avoid reduction in gcc-5\n",
+    "eq0 = Inc(g[0], 1, implicit_dims=(f.dimensions + (ci,)))\n",
+    "\n",
+    "op0 = Operator([eq0])\n",
+    "op0.apply()\n",
+    "\n",
+    "assert (np.count_nonzero(f.data) == g.data[0])\n",
+    "\n",
+    "print(op0.ccode)\n",
+    "g.data[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Example B - Find nonzero positions \n",
+    "\n",
+    "Here we present another example of using `ConditionalDimension`. In the following example, an `Operator` counts the nonzero elements of an array.\n",
+    "\n",
+    "Now, let's take it a step further to return the nonzero positions in $f$. We know that we have `g.data[0]` nonzero elements. In this example, we use the Devito DSL to generate code that will save the nonzero positions."
```

Comment:
shuold be \`f\` not $f$ , right?

Reply:
> In this example, we use the Devito DSL to generate code that will save the nonzero positions."

background noise, drop

you already have said what you do in the previous paragraph, ie

"In the following example, an `Operator` counts the nonzero elements of an array."

Reply:
rewritten, hope its better now.